graylog2-server/src/main/java/org/graylog2/bootstrap/preflight/web/resources/CAResource.java

```diff
@@ -88,4 +115,34 @@ public Response uploadCA(@ApiParam(name = "password") @FormDataParam("password")
             return Response.status(Response.Status.BAD_REQUEST).entity(ApiError.create(e.getMessage())).build();
         }
     }
+
+    @POST
+    @Path("clientCert")
+    @NoAuditEvent("No Audit Event needed")
+    @ApiOperation("Creates a client certificate")
+    @Produces(MediaType.APPLICATION_JSON)
+    @RequiresPermissions("*")
+    public Response createClientCert(@ApiParam(name = "request", required = true) @NotNull @Valid CreateClientCertRequest request) {
```

Comment:
we should add audit events and specific permissions for all certificate management functions.

Reply:
I agree, we also have this as a general to-do in our project board because we lack those in other places, too.